### PR TITLE
Fixed the non-zero return code issue in archive path

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
flytectl register for archive file would overwrite the failed error code while removing the temporary directory.
This PR Segregates the error variables .

Before the fix.
```
flytectl register file  fix.tgz  -d development  -p flytectldemo --version v1 --archive
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| NAME (7)                                                                                                                                | STATUS | ADDITIONAL INFO                                       |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register986812347/Users/praful/flytesnacks/cookbook/core/_pb_output/0_core.scheduled_workflows.lp_schedules.format_date_1.pb       | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register986812347/Users/praful/flytesnacks/cookbook/core/_pb_output/1_core.scheduled_workflows.lp_schedules.date_formatter_wf_2.pb | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register986812347/Users/praful/flytesnacks/cookbook/core/_pb_output/2_core.scheduled_workflows.lp_schedules.date_formatter_wf_3.pb | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register986812347/Users/praful/flytesnacks/cookbook/core/_pb_output/3_my_cron_scheduled_lp_3.pb                                    | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register986812347/Users/praful/flytesnacks/cookbook/core/_pb_output/4_core.scheduled_workflows.lp_schedules.be_positive_1.pb       | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register986812347/Users/praful/flytesnacks/cookbook/core/_pb_output/5_core.scheduled_workflows.lp_schedules.positive_wf_2.pb       | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register986812347/Users/praful/flytesnacks/cookbook/core/_pb_output/6_core.scheduled_workflows.lp_schedules.positive_wf_3.pb       | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
7 rows
(base) ➜  flytectl git:(pmahindrakar/fix-register-return) ✗ echo $?
0

```

After the fix
```
flytectl register file  fix.tgz  -d development  -p flytectldemo --version v1 --archive
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| NAME (7)                                                                                                                                | STATUS | ADDITIONAL INFO                                       |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register777340964/Users/praful/flytesnacks/cookbook/core/_pb_output/0_core.scheduled_workflows.lp_schedules.format_date_1.pb       | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register777340964/Users/praful/flytesnacks/cookbook/core/_pb_output/1_core.scheduled_workflows.lp_schedules.date_formatter_wf_2.pb | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register777340964/Users/praful/flytesnacks/cookbook/core/_pb_output/2_core.scheduled_workflows.lp_schedules.date_formatter_wf_3.pb | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register777340964/Users/praful/flytesnacks/cookbook/core/_pb_output/3_my_cron_scheduled_lp_3.pb                                    | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register777340964/Users/praful/flytesnacks/cookbook/core/_pb_output/4_core.scheduled_workflows.lp_schedules.be_positive_1.pb       | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register777340964/Users/praful/flytesnacks/cookbook/core/_pb_output/5_core.scheduled_workflows.lp_schedules.positive_wf_2.pb       | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
| /tmp/register777340964/Users/praful/flytesnacks/cookbook/core/_pb_output/6_core.scheduled_workflows.lp_schedules.positive_wf_3.pb       | Failed | Error registering file due to rpc error: code =       |
|                                                                                                                                         |        | Unauthenticated desc = token parse error              |
|                                                                                                                                         |        | [JWT_VERIFICATION_FAILED] Could not retrieve id token |
|                                                                                                                                         |        | from metadata, caused by: rpc error: code =           |
|                                                                                                                                         |        | Unauthenticated desc = Request unauthenticated with   |
|                                                                                                                                         |        | IDToken                                               |
 ----------------------------------------------------------------------------------------------------------------------------------------- -------- ------------------------------------------------------- 
7 rows
Error: rpc error: code = Unauthenticated desc = token parse error [JWT_VERIFICATION_FAILED] Could not retrieve id token from metadata, caused by: rpc error: code = Unauthenticated desc = Request unauthenticated with IDToken
(base) ➜  flytectl git:(pmahindrakar/fix-register-return) ✗ echo $?                                                                                
1

```

## Type
- [X] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [X] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by pradithya** [comment]
Thanks @pmahindrakar-oss 
Can you add a unit test for this?

**Comment by pmahindrakar-oss** [comment]
Done. @pradithya .

**Review by pradithya — APPROVED** [review]
LGTM!
